solr/solr-ref-guide/modules/deployment-guide/pages/node-roles.adoc

```diff
@@ -66,15 +69,37 @@ A node with this role (in mode "on") can host shards and replicas for collection
 === `overseer` role
 A node with this role can perform duties of an overseer node (unless mode is `disallowed`). When one or more nodes have the overseer role in `preferred` mode, the overseer leader will be elected from one of these nodes. In case no node is designated as a preferred overseer or no such node is live, the overseer leader will be elected from one of the nodes that have the overseer role in `allowed` mode. If all nodes that are designated with overseer role (allowed or preferred) are down, the cluster will be left without an overseer.
 
+=== `coordinator` role
+
+A node with this role can act as if it has replicas of all collections in the cluster when a query is performed. The workflow is as follows
+
+If the cluster has collections with very large no:of shards, performing distributed requests in your _`data node`_ will lead to
+
+* large heap utilization
+* frequent GC pauses
+
+In such cases, a few dedicated nodes can be started with a *`coordinator`* role and queries can be sent to that node and avoid intermittent and unpredictable load in data nodes. The coordinator node is stateless and does not host any data. So, we can create and destroy coordinator nodes without any data loass or down time.
+
+==== The work-flow in a `coordinator` node
+
+1. A request for *`coll-A`* that uses configset *`configset-A`* comes to coordinator node
+2. It checks if there is a core that uses the configset *`configset-A`* is present. If yes, that core acts as a replica of *`coll-A`* and performs a distributed request to all shards of *`coll-A`* and sends back a response
```

Comment:
This implies that if `collection-1` and `collection-2` are both using the same `configset` then a single synthetic core would be able to service requests for the both of them. I don't think that is accurate from my reading of the code in `CoordinatorHttpSolrCall.getCore` - we should either be more precise with the docs or improve the code to handle this case.

Reply:
> This implies that if collection-1 and collection-2 are both using the same configset then a single synthetic core would be able to service requests for the both of them

yes. why do you think otherwise ?